sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -127,37 +126,34 @@ public OpenAIClient(string openAIApiKey)
         }
 
         /// <summary> Return textual completions as configured for a given prompt. </summary>
-        /// <param name="deploymentOrModelName">
-        ///     Specifies either the model deployment name (when using Azure OpenAI) or model name (when using
-        ///     non-Azure OpenAI) to use for this request.
-        /// </param>
         /// <param name="completionsOptions">
         ///     The options for this completions request.
         /// </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException">
-        ///     <paramref name="deploymentOrModelName"/> or <paramref name="completionsOptions"/> is null.
+        ///     <paramref name="completionsOptions"/> is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///     <paramref name="completionsOptions.DeploymentName"/> is null or empty.
         /// </exception>
```

Comment:
If `completionsOptions.DeploymentName` is null, this would be an `ArgumentNullException` instead of an `ArgumentException`. Same in the other methods.

Reply:
Thank you!! Did a consistency pass across all the XML comments for this *and* the actual use, which also caught missing `Argument.AssertNot*` calls in `EmbeddingsOptions` as well as an incomplete `NotNull` for `CompletionsOptions`.